.github/workflows/test_and_deploy.yml

```
@@ -38,39 +38,83 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      # these libraries enable testing on Qt on linux
+      # These libraries enable testing on Qt on linux
       - uses: tlambert03/setup-qt-libs@v1
 
-      # strategy borrowed from vispy for installing opengl libs on windows
+      # Strategy borrowed from vispy for installing opengl libs on Windows
       - name: Install Windows OpenGL
         if: runner.os == 'Windows'
         run: |
           git clone --depth 1 https://github.com/pyvista/gl-ci-helpers.git
           powershell gl-ci-helpers/appveyor/install_opengl.ps1
           if (Test-Path -Path "C:\Windows\system32\opengl32.dll" -PathType Leaf) {Exit 0} else {Exit 1}
 
-      # note: if you need dependencies from conda, considering using
-      # setup-miniconda: https://github.com/conda-incubator/setup-miniconda
-      # and
-      # tox-conda: https://github.com/tox-dev/tox-conda
-      - name: Install dependencies
+      # Set up Miniconda
+      - name: Set up Miniconda
+        uses: conda-incubator/setup-miniconda@v3
+        with:
+          miniconda-version: "latest"
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
+
+      # Initialize Conda on Linux/macOS
+      - name: Initialize Conda (Linux/macOS)
+        if: runner.os != 'Windows'
+        shell: bash -l {0}
+        run: |
+          conda init bash
+
+      - name: Create and activate environment (Linux/macOS)
+        if: runner.os != 'Windows'
+        shell: bash -l {0}
         run: |
+          conda create --name test-env python=${{ matrix.python-version }} -y
+          source activate test-env
```

I think this should fix the failing activate

```suggestion
          conda activate test-env
```